File: guides/micronaut-security-session-database-authentication/java/src/main/java/example/micronaut/controllers/HomeController.java
/*
 * Copyright 2017-2023 original authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package example.micronaut.controllers;

import io.micronaut.http.MediaType;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.Produces;
import io.micronaut.security.annotation.Secured;
import io.micronaut.security.rules.SecurityRule;
import io.micronaut.views.View;

import java.util.Collections;
import java.util.Map;

@Controller // <1>
public class HomeController {

    @Produces(MediaType.TEXT_HTML) // <2>
    @Secured(SecurityRule.IS_ANONYMOUS) // <3>
    @Get // <4>
    @View("home.html") // <5>
    Map<String, Object> index() {
        return Collections.emptyMap();
    }
}

File: guides/micronaut-security-session-database-authentication/python/src/example/micronaut/controllers/home_controller.py
from micronaut.http import MediaType
from micronaut.http.annotation import Get, Produces
from micronaut.security.annotation import Secured
from micronaut.security.rules import SecurityRule
from micronaut.views import View


# tag::clazz[]
@Produces(MediaType.TEXT_HTML)  # <1>
@Secured(SecurityRule.IS_ANONYMOUS)  # <2>
@Get("/")  # <3>
@View("home.html")  # <4>
def index() -> dict:  # <5>
    return {}
# end::clazz[]
